### ISSUE TITLE
val/cls_loss=inf

### ISSUE THREAD

**Issue by cronousbaby** [ultralytics/ultralytics#20439]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

yolo11 obj-det  happen:val/cls_loss=inf
my env:
ultralytics              8.3.122
ultralytics-thop         2.0.14
this is train record:
epoch,time,train/box_loss,train/cls_loss,train/dfl_loss,metrics/precision(B),metrics/recall(B),metrics/mAP50(B),metrics/mAP50-95(B),val/box_loss,val/cls_loss,val/dfl_loss,lr/pg0,lr/pg1,lr/pg2
1,55.3016,1.6386,5.24482,1.12943,0,0,0,0,0.78415,inf,0.84331,7.29e-05,7.29e-05,7.29e-05
2,114.718,0.75318,4.93818,0.8687,0.03211,0.00325,0.0165,0.01419,0.57505,inf,0.8402,0.00014801,0.00014801,0.00014801
3,194.94,0.56703,4.61053,0.83383,0.04727,0.1327,0.02841,0.02462,0.44947,inf,0.82042,0.000222621,0.000222621,0.000222621
4,283.709,0.52265,4.10182,0.82527,0.04375,0.66004,0.0834,0.07432,0.39905,inf,0.8155,0.000267327,0.000267327,0.000267327
5,374.712,0.50444,3.26608,0.82189,0.15074,0.2976,0.16668,0.15502,0.34114,inf,0.80681,0.000266436,0.000266436,0.000266436
6,465.266,0.47747,2.58149,0.81684,0.17191,0.49472,0.23716,0.2243,0.33118,inf,0.80572,0.000265545,0.000265545,0.000265545
7,557.375,0.44405,2.21069,0.81286,0.26548,0.57045,0.34924,0.33425,0.3107,inf,0.80888,0.000264654,0.000264654,0.000264654
8,648.912,0.42048,1.95168,0.81031,0.37807,0.61175,0.49024,0.47005,0.29777,inf,0.81047,0.000263763,0.000263763,0.000263763
9,738.758,0.39795,1.71114,0.80881,0.50176,0.71228,0.63366,0.61183,0.27966,inf,0.81262,0.000262872,0.000262872,0.000262872
10,828.367,0.38298,1.48992,0.80748,0.56577,0.77901,0.70621,0.68465,0.25979,inf,0.81367,0.000261981,0.000261981,0.000261981
11,919.4,0.36485,1.30875,0.80448,0.64054,0.80721,0.74752,0.72307,0.27121,inf,0.81584,0.00026109,0.00026109,0.00026109
12,1011.35,0.35406,1.19469,0.80384,0.6937,0.87304,0.77164,0.75154,0.24603,inf,0.81559,0.000260199,0.000260199,0.000260199
13,1103.44,0.34232,1.11776,0.80138,0.72635,0.91307,0.79324,0.77292,0.24932,inf,0.81508,0.000259308,0.000259308,0.000259308
14,1194.55,0.33318,1.06705,0.80009,0.7544,0.91957,0.81942,0.80008,0.23977,inf,0.8142,0.000258417,0.000258417,0.000258417
15,1282.77,0.32661,1.01943,0.79909,0.79139,0.90599,0.87212,0.85292,0.23855,inf,0.8126,0.000257526,0.000257526,0.000257526
16,1374.73,0.31843,0.95962,0.79834,0.83694,0.90216,0.91604,0.89961,0.22451,inf,0.81235,0.000256635,0.000256635,0.000256635
17,1465.8,0.31722,0.90366,0.79715,0.87725,0.91156,0.93903,0.92455,0.21722,inf,0.81203,0.000255744,0.000255744,0.000255744
18,1557.28,0.31044,0.85104,0.79584,0.90179,0.93035,0.95374,0.93953,0.21664,inf,0.81254,0.000254853,0.000254853,0.000254853
19,1648.75,0.30417,0.80468,0.79504,0.91383,0.9321,0.95937,0.94619,0.21121,inf,0.81231,0.000253962,0.000253962,0.000253962
20,1741.25,0.29801,0.76736,0.79475,0.92642,0.93885,0.96292,0.94822,0.21923,inf,0.81258,0.000253071,0.000253071,0.000253071

### Environment

Ultralytics 8.3.122 🚀 Python-3.12.9 torch-2.5.1+cu124

### Minimal Reproducible Example

yolo11 obj-det  happen:val/cls_loss=inf 

yolo11 obj-det  happen:val/cls_loss=inf
my env:
ultralytics              8.3.122
ultralytics-thop         2.0.14
this is train record:
epoch,time,train/box_loss,train/cls_loss,train/dfl_loss,metrics/precision(B),metrics/recall(B),metrics/mAP50(B),metrics/mAP50-95(B),val/box_loss,val/cls_loss,val/dfl_loss,lr/pg0,lr/pg1,lr/pg2
1,55.3016,1.6386,5.24482,1.12943,0,0,0,0,0.78415,inf,0.84331,7.29e-05,7.29e-05,7.29e-05
2,114.718,0.75318,4.93818,0.8687,0.03211,0.00325,0.0165,0.01419,0.57505,inf,0.8402,0.00014801,0.00014801,0.00014801
3,194.94,0.56703,4.61053,0.83383,0.04727,0.1327,0.02841,0.02462,0.44947,inf,0.82042,0.000222621,0.000222621,0.000222621
4,283.709,0.52265,4.10182,0.82527,0.04375,0.66004,0.0834,0.07432,0.39905,inf,0.8155,0.000267327,0.000267327,0.000267327
5,374.712,0.50444,3.26608,0.82189,0.15074,0.2976,0.16668,0.15502,0.34114,inf,0.80681,0.000266436,0.000266436,0.000266436
6,465.266,0.47747,2.58149,0.81684,0.17191,0.49472,0.23716,0.2243,0.33118,inf,0.80572,0.000265545,0.000265545,0.000265545
7,557.375,0.44405,2.21069,0.81286,0.26548,0.57045,0.34924,0.33425,0.3107,inf,0.80888,0.000264654,0.000264654,0.000264654
8,648.912,0.42048,1.95168,0.81031,0.37807,0.61175,0.49024,0.47005,0.29777,inf,0.81047,0.000263763,0.000263763,0.000263763
9,738.758,0.39795,1.71114,0.80881,0.50176,0.71228,0.63366,0.61183,0.27966,inf,0.81262,0.000262872,0.000262872,0.000262872
10,828.367,0.38298,1.48992,0.80748,0.56577,0.77901,0.70621,0.68465,0.25979,inf,0.81367,0.000261981,0.000261981,0.000261981
11,919.4,0.36485,1.30875,0.80448,0.64054,0.80721,0.74752,0.72307,0.27121,inf,0.81584,0.00026109,0.00026109,0.00026109
12,1011.35,0.35406,1.19469,0.80384,0.6937,0.87304,0.77164,0.75154,0.24603,inf,0.81559,0.000260199,0.000260199,0.000260199
13,1103.44,0.34232,1.11776,0.80138,0.72635,0.91307,0.79324,0.77292,0.24932,inf,0.81508,0.000259308,0.000259308,0.000259308
14,1194.55,0.33318,1.06705,0.80009,0.7544,0.91957,0.81942,0.80008,0.23977,inf,0.8142,0.000258417,0.000258417,0.000258417
15,1282.77,0.32661,1.01943,0.79909,0.79139,0.90599,0.87212,0.85292,0.23855,inf,0.8126,0.000257526,0.000257526,0.000257526
16,1374.73,0.31843,0.95962,0.79834,0.83694,0.90216,0.91604,0.89961,0.22451,inf,0.81235,0.000256635,0.000256635,0.000256635
17,1465.8,0.31722,0.90366,0.79715,0.87725,0.91156,0.93903,0.92455,0.21722,inf,0.81203,0.000255744,0.000255744,0.000255744
18,1557.28,0.31044,0.85104,0.79584,0.90179,0.93035,0.95374,0.93953,0.21664,inf,0.81254,0.000254853,0.000254853,0.000254853
19,1648.75,0.30417,0.80468,0.79504,0.91383,0.9321,0.95937,0.94619,0.21121,inf,0.81231,0.000253962,0.000253962,0.000253962
20,1741.25,0.29801,0.76736,0.79475,0.92642,0.93885,0.96292,0.94822,0.21923,inf,0.81258,0.000253071,0.000253071,0.000253071

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cronousbaby, thank you for reporting this and for providing detailed logs and environment info! 🚀 This is an automated response to help you get the fastest possible assistance—an Ultralytics engineer will review and assist you soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for guidance on training and troubleshooting, especially the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. If you haven't already, please check our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for advice on preparing your data and training.

If this is a 🐛 Bug Report, please ensure you provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us pinpoint the issue. The information you've provided is very helpful—if there are any additional details about your data, configuration, or training command, please include those as well.

Join the Ultralytics community for support and discussion:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and learn with the community

## Upgrade

Please make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue is not already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO runs in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for helping us improve Ultralytics! 💡

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

Can you provide your training command?

**Comment by cronousbaby** [comment]
@Y-T-G 
 task=detect, mode=train, model=weights/yolo11n.pt, data=dataset.yaml, epochs=300, time=None, patience=50, batch=3600, imgsz=192, save=True, save_period=-1, cache=ram, device=0,1,2,3, workers=40, project=my_detection, name=yolov11n_test, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=42, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, cutmix=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, cfg=None, tracker=botsort.yaml, save_dir=my_detection/test
Overriding model.yaml nc=80 with nc=33

**Comment by Y-T-G** [comment]
Post only the arguments you passed in your command. Not this. It's hard to tell the arguments you used from this.

**Comment by cronousbaby** [comment]
@Y-T-G   ok ! 
yolo detect train \
  model=weights/yolov11s.pt \
  data=dataset.yaml \
  epochs=300 \
  patience=50 \
  imgsz=192 \
  batch=2400 \
  workers=40 \
  cache=ram \
  device=0,1,2,3 \
  project=map_detection \
  name=yolov11s_test \
  verbose=True \
  plots=True \
  seed=42 \
  deterministic=True \
  multi_scale=False

**Comment by Y-T-G** [comment]
Your `batch` is too high

**Comment by cronousbaby** [comment]
@Y-T-G  Thank you for the feedback regarding the high batch size potentially causing the val/cls_loss=inf issue I reported
However, the key observation is that despite the val/cls_loss being inf, other crucial validation metrics, particularly metrics/mAP50-95(B), improved steadily and reached very high values (saturating around 0.991 for the Nano model before early stopping at epoch 139). This suggests the model itself was learning effectively and making accurate predictions on the validation set, as reflected by the mAP.
This leads me to wonder if the high batch size is the direct cause of an invalid loss calculation itself, or perhaps if it's indirectly triggering a numerical instability issue (maybe related to AMP/FP16 at extreme values?) or a potential bug specifically within the validation classification loss computation or logging part of the framework? It seems strange that the mAP calculation remains unaffected if the classification loss is truly infinite. I also noticed similar val/cls_loss=inf reports in previous community discussions.
Could this inf value potentially stem from a numerical precision issue (e.g., with AMP) or a logging quirk triggered specifically by very large batches, rather than indicating a fundamental training failure?
I will test with a significantly lower batch_per_gpu (like 1024-1536) as advised and check if val/cls_loss normalizes. Any further insight you might have on why this specific loss value becomes inf even with good mAP would be greatly appreciated.
thanks!

**Comment by Y-T-G** [comment]
The train cls loss is fine, so it will still learn since that's what is used for backpropagation. Validation loss isn't used for learning.

Also validation runs at double the batch size. So when you use 3000, validation will run at 6000.
